### PR TITLE
Change Travis to use Focal Fossa and Postgresql 12

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] CI improvement

## Description
Default Travis build uses Ubuntu 16. This changes it to 20.04 so we get all the benefits of using the newer OS. Along with that, update the PSQL version.

## Related Tickets & Documents
Shoutout to @djuber for helping me out.

References
- https://alphahydrae.com/2021/02/how-to-run-postgresql-13-on-travis-ci/
- https://docs.travis-ci.com/user/database-setup/#using-a-different-postgresql-version
## Added tests?
- [x] no, because they aren't needed

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,6 +43,13 @@ env:
     - COVERAGE_REPORTS_TOTAL=4
     - FOREM_OWNER_SECRET="secret" # test secret so e2e tests can run properly.
 before_install:
+  - >-
+    sudo sed -i
+    -e '/local.*peer/s/postgres/all/'
+    -e 's/peer\|md5/trust/g'
+    /etc/postgresql/12/main/pg_hba.conf
+  # Restart the PostgreSQL service:
```
Comment (maestromac) [inline .travis.yml:51]:
```suggestion
```

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +10,11 @@ cache:
 rvm:
   - 2.7.2
 addons:
-  postgresql: '10' # Travis does not work out of the box with Postgres 11 yet
+  postgresql: '12'
+  apt:
+    packages:
+      - postgresql-12
+      - postgresql-client-12
```
Comment (maestromac) [inline .travis.yml:17]: With v12, Travis requires quite a bit of tweaking.

File: .travis.yml [hunk 3]
```diff
@@ -38,6 +43,12 @@ env:
     - COVERAGE_REPORTS_TOTAL=4
     - FOREM_OWNER_SECRET="secret" # test secret so e2e tests can run properly.
 before_install:
+  - >-
+    sudo sed -i
+    -e '/local.*peer/s/postgres/all/'
+    -e 's/peer\|md5/trust/g'
+    /etc/postgresql/12/main/pg_hba.conf
+  - sudo service postgresql@12-main restart
```
Comment (maestromac) [inline .travis.yml:51]: Change permission of withing `pg_hba.conf` so we can continue to access PG without a password

**Review by jgaskins — APPROVED** [review]

**Review by rhymes — APPROVED** [review]

**Comment by rhymes** [comment]
Yes!

**Pull request merged**